modules/msk/custom/splitfastqbylane/main.nf

```diff
@@ -0,0 +1,45 @@
+process CUSTOM_SPLITFASTQBYLANE {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/gawk:5.1.0':
+        'biocontainers/gawk:5.1.0' }"
+
+    input:
+    tuple val(meta), path(reads)
+
+    output:
+    tuple val(meta), path("*.split.fastq.gz"), emit: reads
+    path "versions.yml"                      , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def read1 = [reads].flatten()[0]
+    def read2 = [reads].flatten().size() > 1 ? reads[1] : null
+    """
+    split_lanes_awk.sh ${prefix} ${read1} ${read2 ?: ''}
```

Comment:
Could you remove the `${read2 ?: ''}` and just do `${read2}` since you already set it to null if the second read item is empty? Or if that doesn't work would this: 
```
    def read2 = [reads].flatten().size() > 1 ? reads[1] : ''
    """
    split_lanes_awk.sh ${prefix} ${read1} ${read2}
```
Just a syntactical note, no worries if this is best.

Reply:
oh that makes total sense. yea i'll fix that.